## 1. The oldest business in the world
![The entrance to St. Peter Stiftskeller, a restaurant in Saltzburg, Austria. The sign over the entrance shows "803" - the year the business opened.](https://assets.datacamp.com/production/repositories/5851/datasets/7dded924c6dc418d4a828f2f4daba99953c27a5a/400px-Eingang_zum_St._Peter_Stiftskeller.jpg)  
*Image: St. Peter Stiftskeller, founded 803. Credit: [Pakeha](https://commons.wikimedia.org/wiki/File:Eingang_zum_St._Peter_Stiftskeller.jpg).*

An important part of business is planning for the future and ensuring that the company survives changing market conditions. Some businesses do this really well and last for hundreds of years.

BusinessFinancing.co.uk [researched](https://businessfinancing.co.uk/the-oldest-company-in-almost-every-country) the oldest company that is still in business in (almost) every country and compiled the results into a dataset. In this project, you'll explore that dataset to see what they found.

The database contains three tables.

### `categories`

| column         | type    | meaning                            |
|----------------|---------|------------------------------------|
| category_code  | varchar | Code for the category of the business |
| category       | varchar | Description of the business category |

### `countries`

| column         | type    | meaning                            |
|----------------|---------|------------------------------------|
| country_code   | varchar | ISO 3166-1 3-letter country code    |
| country        | varchar | Name of the country                 |
| continent      | varchar | Name of the continent that the country exists in |

### `businesses`

| column         | type    | meaning                            |
|----------------|---------|------------------------------------|
| business       | varchar | Name of the business                |
| year_founded   | int     | Year the business was founded       |
| category_code  | varchar | Code for the category of the business |
| country_code   | char    | ISO 3166-1 3-letter country code    |

Let's begin by looking at the range of the founding years throughout the world.


In [67]:
%%sql 
postgresql:///oldestbusinesses

SELECT 
    MIN(year_founded), 
    MAX(year_founded)
FROM businesses;

1 rows affected.


[(578, 1999)]

## 2. How many businesses were founded before 1000?
<p>Wow! That's a lot of variation between countries. In one country, the oldest business was only founded in 1999. By contrast, the oldest business in the world was founded back in 578. That's pretty incredible that a business has survived for more than a millennium.</p>
<p>I wonder how many other businesses there are like that.</p>

In [69]:
%%sql

SELECT COUNT(*)
FROM businesses
WHERE year_founded < 1000;

 * postgresql:///oldestbusinesses
1 rows affected.


[(6,)]

## 3. Which businesses were founded before 1000?
<p>Having a count is all very well, but I'd like more detail. Which businesses have been around for more than a millennium?</p>

In [71]:
%%sql

SELECT *
FROM businesses
WHERE year_founded < 1000
ORDER BY year_founded;

 * postgresql:///oldestbusinesses
6 rows affected.


[('Kongō Gumi', 578, 'CAT6', 'JPN'),
 ('St. Peter Stifts Kulinarium', 803, 'CAT4', 'AUT'),
 ('Staffelter\xa0Hof Winery', 862, 'CAT9', 'DEU'),
 ('Monnaie de Paris\xa0', 864, 'CAT12', 'FRA'),
 ('The Royal Mint', 886, 'CAT12', 'GBR'),
 ("Sean's Bar", 900, 'CAT4', 'IRL')]

## 4. Exploring the categories
<p>Now we know that the oldest, continuously operating company in the world is called Kongō Gumi. But was does that company do? The category codes in the <code>businesses</code> table aren't very helpful: the descriptions of the categories are stored in the <code>categories</code> table.</p>
<p>This is a common problem: for data storage, it's better to keep different types of data in different tables, but for analysis, you want all the data in one place. To solve this, you'll have to join the two tables together.</p>

In [73]:
%%sql

SELECT
    business,
    year_founded,
    country_code,
    category
FROM businesses
INNER JOIN categories
    USING(category_code)
WHERE year_founded < 1000
ORDER BY year_founded;

 * postgresql:///oldestbusinesses
6 rows affected.


[('Kongō Gumi', 578, 'JPN', 'Construction'),
 ('St. Peter Stifts Kulinarium', 803, 'AUT', 'Cafés, Restaurants & Bars'),
 ('Staffelter\xa0Hof Winery', 862, 'DEU', 'Distillers, Vintners, & Breweries'),
 ('Monnaie de Paris\xa0', 864, 'FRA', 'Manufacturing & Production'),
 ('The Royal Mint', 886, 'GBR', 'Manufacturing & Production'),
 ("Sean's Bar", 900, 'IRL', 'Cafés, Restaurants & Bars')]

## 5. Counting the categories
<p>With that extra detail about the oldest businesses, we can see that Kongō Gumi is a construction company. In that list of six businesses, we also see a café, a winery, and a bar. The two companies recorded as "Manufacturing and Production" are both mints. That is, they produce currency.</p>
<p>I'm curious as to what other industries constitute the oldest companies around the world, and which industries are most common.</p>

In [75]:
%%sql


SELECT
    category,
    COUNT(*) AS n
FROM categories
INNER JOIN businesses
    USING(category_code)
GROUP BY category
ORDER BY n desc
LIMIT 10;

 * postgresql:///oldestbusinesses
10 rows affected.


[('Banking & Finance', 37),
 ('Distillers, Vintners, & Breweries', 22),
 ('Aviation & Transport', 19),
 ('Postal Service', 16),
 ('Manufacturing & Production', 15),
 ('Media', 7),
 ('Agriculture', 6),
 ('Cafés, Restaurants & Bars', 6),
 ('Food & Beverages', 6),
 ('Tourism & Hotels', 4)]

## 6. Oldest business by continent
<p>It looks like "Banking &amp; Finance" is the most popular category. Maybe that's where you should aim if you want to start a thousand-year business.</p>
<p>One thing we haven't looked at yet is where in the world these really old businesses are. To answer these questions, we'll need to join the <code>businesses</code> table to the <code>countries</code> table. Let's start by asking how old the oldest business is on each continent.</p>

In [77]:
%%sql

SELECT
    MIN(year_founded) as oldest,
    continent
FROM businesses
INNER JOIN countries
    USING(country_code)
GROUP BY continent
ORDER BY oldest ASC;

 * postgresql:///oldestbusinesses
6 rows affected.


[(578, 'Asia'),
 (803, 'Europe'),
 (1534, 'North America'),
 (1565, 'South America'),
 (1772, 'Africa'),
 (1809, 'Oceania')]

## 7. Joining everything for further analysis
<p>Interesting. There's a jump in time from the older businesses in Asia and Europe to the 16th Century oldest businesses in North and South America, then to the 18th and 19th Century oldest businesses in Africa and Oceania. </p>
<p>As mentioned earlier, when analyzing data it's often really helpful to have all the tables you want access to joined together into a single set of results that can be analyzed further. Here, that means we need to join all three tables.</p>

In [79]:
%%sql

SELECT
    business,
    year_founded,
    category,
    continent
FROM businesses
INNER JOIN categories
    USING(category_code)
INNER JOIN countries
    USING(country_code);


 * postgresql:///oldestbusinesses
163 rows affected.


[('Spinzar Cotton Company', 1930, 'Agriculture', 'Asia'),
 ('ALBtelecom', 1912, 'Telecommunications', 'Europe'),
 ('Andbank', 1930, 'Banking & Finance', 'Europe'),
 ('Liwa Chemicals', 1939, 'Manufacturing & Production', 'Asia'),
 ('Bank of the Province of Buenos Aires', 1822, 'Banking & Finance', 'South America'),
 ('Yerevan Ararat Brandy-Wine-Vodka Factory', 1877, 'Distillers, Vintners, & Breweries', 'Asia'),
 ('Australia Post', 1809, 'Postal Service', 'Oceania'),
 ('St. Peter Stifts Kulinarium', 803, 'Cafés, Restaurants & Bars', 'Europe'),
 ('Azerbaijan Caspian Shipping Company', 1858, 'Aviation & Transport', 'Asia'),
 ('Brarudi', 1955, 'Distillers, Vintners, & Breweries', 'Africa'),
 ('Affligem Brewery', 1074, 'Distillers, Vintners, & Breweries', 'Europe'),
 ('Communauté Électrique du Bénin', 1968, 'Energy', 'Africa'),
 ('Air Burkina', 1967, 'Aviation & Transport', 'Africa'),
 ('M. M. Ispahani Limited', 1820, 'Food & Beverages', 'Asia'),
 ('Arsenal AD', 1878, 'Defense', 'Europe'),
 ('BMMI', 1883, 'Retail', 'Asia'),
 ('Sarajevska Pivara', 1864, 'Distillers, Vintners, & Breweries', 'Europe'),
 ('Olivaria Brewery', 1864, 'Distillers, Vintners, & Breweries', 'Europe'),
 ('Belize Bank', 1902, 'Banking & Finance', 'North America'),
 ('Banco Nacional de Bolivia', 1871, 'Banking & Finance', 'South America'),
 ('Casa da Moeda do Brasil', 1694, 'Manufacturing & Production', 'South America'),
 ('Mount Gay Rum', 1703, 'Distillers, Vintners, & Breweries', 'North America'),
 ('Hua Ho Department Store', 1947, 'Retail', 'Asia'),
 ('Tashi Group', 1959, 'Aviation & Transport', 'Asia'),
 ('Botswana Meat Commission', 1965, 'Agriculture', 'Africa'),
 ('Banque Internationale pour la Centrafrique', 1946, 'Banking & Finance', 'Africa'),
 ("Hudson's Bay Company", 1670, 'Retail', 'North America'),
 ('Gasthof Sternen', 1230, 'Tourism & Hotels', 'Europe'),
 ('Famae', 1811, 'Defense', 'South America'),
 ("Ma Yu Ching's Bucket Chicken House", 1153, 'Cafés, Restaurants & Bars', 'Asia'),
 ('La Poste', 1945, 'Postal Service', 'Africa'),
 ('Cameroon Development Corporation', 1947, 'Agriculture', 'Africa'),
 ('Société nationale des Chemins de fer du Congo', 1889, 'Aviation & Transport', 'Africa'),
 ('Development Bank of the Central African States', 1975, 'Banking & Finance', 'Africa'),
 ('Casa de Moneda de Colombia', 1621, 'Manufacturing & Production', 'South America'),
 ('Central Bank of the Comoros', 1981, 'Banking & Finance', 'Africa'),
 ('Correios de Cabo Verde', 1849, 'Postal Service', 'Africa'),
 ('Florida Ice and Farm Company', 1908, 'Distillers, Vintners, & Breweries', 'North America'),
 ('Cubana de Aviación', 1929, 'Aviation & Transport', 'North America'),
 ('Bank of Cyprus', 1899, 'Banking & Finance', 'Europe'),
 ('Pivovar Broumov', 1348, 'Distillers, Vintners, & Breweries', 'Europe'),
 ('Staffelter\xa0Hof Winery', 862, 'Distillers, Vintners, & Breweries', 'Europe'),
 ('Ethio-Djibouti Railways', 1901, 'Aviation & Transport', 'Africa'),
 ('The Chronicle (Dominica)', 1909, 'Media', 'North America'),
 ('Munke Mølle', 1135, 'Manufacturing & Production', 'Europe'),
 ('Hamoud Boualem', 1878, 'Food & Beverages', 'Africa'),
 ('Egyptian National Railways', 1854, 'Aviation & Transport', 'Africa'),
 ('Asmara Brewery', 1939, 'Distillers, Vintners, & Breweries', 'Africa'),
 ('Casa de Ganaderos', 1218, 'Agriculture', 'Europe'),
 ('Raeapteek', 1422, 'Medical', 'Europe'),
 ('National Bank of Ethiopia', 1906, 'Banking & Finance', 'Africa'),
 ('Fiskars', 1649, 'Consumer Goods', 'Europe'),
 ('Monnaie de Paris\xa0', 864, 'Manufacturing & Production', 'Europe'),
 ('BGFIBank Group', 1971, 'Banking & Finance', 'Africa'),
 ('The Royal Mint', 886, 'Manufacturing & Production', 'Europe'),
 ('Bank of Georgia', 1903, 'Banking & Finance', 'Asia'),
 ('Halco Mining', 1962, 'Mining', 'Africa'),
 ('Correios da Guiné-Bissau', 1973, 'Postal Service', 'Africa'),
 ('Guinea Ecuatorial Airlines', 1996, 'Aviation & Transport', 'Africa'),
 ('Piraeus Bank', 1606, 'Banking & Finance', 'Euro

## 8. Counting categories by continent
<p>Having <code>businesses</code> joined to <code>categories</code> and <code>countries</code> together means we can ask questions about both these things together. For example, which are the most common categories for the oldest businesses on each continent?</p>

In [81]:
%%sql

SELECT
    continent,
    category,
    COUNT(business) AS n
FROM businesses
INNER JOIN categories
    USING(category_code)
INNER JOIN countries
    USING(country_code)
GROUP BY continent, category;

 * postgresql:///oldestbusinesses
56 rows affected.


[('North America', 'Banking & Finance', 4),
 ('Oceania', 'Postal Service', 1),
 ('South America', 'Food & Beverages', 2),
 ('Europe', 'Tourism & Hotels', 2),
 ('Asia', 'Media', 1),
 ('Europe', 'Medical', 1),
 ('Asia', 'Defense', 1),
 ('Africa', 'Manufacturing & Production', 1),
 ('Europe', 'Postal Service', 4),
 ('North America', 'Aviation & Transport', 2),
 ('Asia', 'Distillers, Vintners, & Breweries', 2),
 ('South America', 'Banking & Finance', 3),
 ('North America', 'Food & Beverages', 1),
 ('Europe', 'Manufacturing & Production', 8),
 ('Africa', 'Postal Service', 9),
 ('Asia', 'Telecommunications', 1),
 ('Africa', 'Food & Beverages', 1),
 ('Europe', 'Consumer Goods', 3),
 ('Europe', 'Mining', 1),
 ('Oceania', 'Banking & Finance', 2),
 ('Asia', 'Agriculture', 1),
 ('North America', 'Manufacturing & Production', 1),
 ('Africa', 'Mining', 1),
 ('Africa', 'Aviation & Transport', 10),
 ('Asia', 'Construction', 2),
 ('Asia', 'Energy', 3),
 ('Asia', 'Retail', 3),
 ('South America', 'Manufacturing & Production', 2),
 ('Asia', 'Cafés, Restaurants & Bars', 3),
 ('Europe', 'Banking & Finance', 5),
 ('Africa', 'Banking & Finance', 17),
 ('North America', 'Tourism & Hotels', 1),
 ('Europe', 'Cafés, Restaurants & Bars', 2),
 ('Asia', 'Banking & Finance', 6),
 ('South America', 'Defense', 1),
 ('Africa', 'Energy', 1),
 ('Asia', 'Conglomerate', 3),
 ('North America', 'Media', 1),
 ('Europe', 'Agriculture', 1),
 ('Asia', 'Mining', 1),
 ('Asia', 'Aviation & Transport', 7),
 ('Asia', 'Food & Beverages', 2),
 ('North America', 'Distillers, Vintners, & Breweries', 5),
 ('Africa', 'Agriculture', 3),
 ('Africa', 'Distillers, Vintners, & Breweries', 3),
 ('North America', 'Agriculture', 1),
 ('Asia', 'Postal Service', 2),
 ('Europe', 'Defense', 1),
 ('Asia', 'Manufacturing & Production', 3),
 ('Europe', 'Telecommunications', 1),
 ('Europe', 'Distillers, Vintners, & Breweries', 12),
 ('South America', 'Cafés, Restaurants & Bars', 1),
 ('North America', 'Retail', 1),
 ('Africa', 'Media', 4),
 ('Europe', 'Media', 1),
 ('Asia', 'Tourism & Hotels', 1)]

## 9. Filtering counts by continent and category
<p>Combining continent and business category led to a lot of results. It's difficult to see what is important. To trim this down to a manageable size, let's restrict the results to only continent/category pairs with a high count.</p>

In [83]:
%%sql

SELECT
    continent,
    category,
    COUNT(business) AS n
FROM businesses
INNER JOIN categories
    USING(category_code)
INNER JOIN countries
    USING(country_code)
GROUP BY continent, category
HAVING COUNT(business) > 5
ORDER BY n DESC;

 * postgresql:///oldestbusinesses
7 rows affected.


[('Africa', 'Banking & Finance', 17),
 ('Europe', 'Distillers, Vintners, & Breweries', 12),
 ('Africa', 'Aviation & Transport', 10),
 ('Africa', 'Postal Service', 9),
 ('Europe', 'Manufacturing & Production', 8),
 ('Asia', 'Aviation & Transport', 7),
 ('Asia', 'Banking & Finance', 6)]